flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/InputProcessorUtil.java

```diff
@@ -90,6 +91,10 @@ public static CheckpointedInputGate createCheckpointedInputGate(
 			inputGate1.getNumberOfInputChannels() + inputGate2.getNumberOfInputChannels(),
 			taskName,
 			toNotifyOnCheckpoint);
+
+		TaskIOMetricGroup taskIOMetricGroup = toNotifyOnCheckpoint.getEnvironment().getMetricGroup().getIOMetricGroup();
+		taskIOMetricGroup.gauge("checkpointAlignmentTime", barrierHandler::getAlignmentDurationNanos);
```

Comment:
deduplicate this code with `OneInputStreamTask`? 

In master version I think this is also missing
```
			taskIOMetricGroup.gauge("checkpointStartDelayNanos", inputGate::getCheckpointStartDelayNanos);
```

Reply:
>deduplicate this code with OneInputStreamTask
I considered that and actually we can remove the method of `getAlignmentDurationNanos` from `CheckpointedInputGate` to unify the way all in the constructor of `CheckpointBarrierHandler`. The current way of touching minimum change is also for catching up with the release ASAP. I would try that if not very hurry.

Reply:
Let me check the issue of `checkpointStartDelayNanos`

Reply:
After checking the code path, the unification in `OneInputStreamTask` is not strong related to this fix. So I prefer to submitting a hotfix or separate ticket for handling the case of `OneInputStreamTask` in master branch.